### ISSUE TITLE
Display Elements/Models Tag Indicators

### ISSUE THREAD

**Issue by McCoyCOREUSA** [xLightsSequencer/xLights#4016]
Regarding: The new tags on models/display elements in 2023.15.

OVERVIEW: The new tags indicate when the MODEL has an effect on it. This is great. However, some of the logic doesn't follow the same in groups vs sub-models, sub-rows, etc.

SUGGESTION: Make the indicator colors customizable, or use other color than yellow. In the standard non-dark theme, the yellow is very difficult to see. In dark mode, it's displaying in orange. Colors just don't match.

SUGGESTION: The yellow/orange indicator displaying when the ROW ITSELF has effects is good. The Zoom Room discussion brought up the need for a second indicator placed next to the current indicator and it would use a different color (red in my mockups).  The red indicator displays when a group or model has effects WITHIN it, but not on the ROW itself. For example, in a group where an effect is present on a model within the group, the group would display a red indicator to show there are effects within that group. If the group itself (the group row) has an effect, then the yellow/orange indicator is also displayed. This logic would be consistent with groups, sub-models, sub-rows, etc.

MOCK-UP EXPLAINATIONS

1) Indicator displays yellow/orange because the effect is on that row. Correct.

2) Yellow/Orange indicator display on the sub-model as well as the model, but the effect is only on the sub-model. We believe the indicator in this example should show yellow/orange on the 'Servos' row only, but a new red indicator should display on 'Santa Servos' main row to indicate there are effects within it, but this should NOT have the yellow/orange on 'Santa Servos' row because there are no effects on this row itself.

3) Group has yellow/orange indicator because the effect is on the main group row. Correct.

4) Model within this group has an effect, so the yellow/orange indicator shows for that row. Correct. Currently, nothing indicates on the group row itself. However, since there is an effect on a model within the group, we suggest the group display the red indicator. If the group AND a model within the group both had effects, the group row would have both indicators side-by-side as shown in mock-up #5. 

This was an in-depth discussion today in the Zoom Room and I offered to write this up on Github.

- Scott McCoy

![2023-10-03_11-56-30](https://github.com/smeighan/xLights/assets/104024793/52605568-9bcf-40ce-b814-d3c59cb94f9f)

**Comment by keithsw1111** [comment]
I disagree with some of this. I very deliberately do not flag a group with a model with an effect on it IF that model also appears in the model list. To do so would render this feature essentially useless as many groups would then always be flagged.

**Comment by derwin12** [comment]
Hidden "below" is a bit more critical here... hidden on submodels or strands would be ideally to be shown.

**Comment by McCoyCOREUSA** [comment]
Keith, we did discuss the 'most groups would be flagged' issue. The Room felt the logic of having flags work the same for all situations was the way to go. Interestingly, when these tags were first discovered and shown in the Room and we were all experimenting to see how it worked - even a seasoned user said, "looks like this is a bug" when we tested on groups. This is where the discussion lead to perhaps it should work the same for everything, even though we all agreed that most groups would be flagged most of the time. 

Just passing along the discussion from the Room. 

Thank you.

**Comment by McCoyCOREUSA** [comment]
Zoom Room discovered groups does show the flag IF the model doesn't exist in the display elements for a sequence. If the model does exist, the group displays differently with no flag.

**Comment by cybercop23** [comment]
This would be especially useful with some of the pro sequences that have tons of layers but for some strange reason the top layer is blank. I tend to collapse all layers since some can get to 20+ to give me more room to see the "whole" list.
If the two-column indication is troublesome, an option to include all, will be a good compromise.

**Comment by robfallone** [comment]
I have opened a PR with changes that I believe are useful and in line with this "enhancement". https://github.com/smeighan/xLights/pull/4071

**Comment by AzGilrock** [comment]
This is exactly why the xLights GUI is no longer as snappy as I had it running.   You realize it slows down rendering the scene every time some new gadget calculations get added.   I’d personally rip all this crap out.

**Comment by robfallone** [comment]
> This is exactly why the xLights GUI is no longer as snappy as I had it running. You realize it slows down rendering the scene every time some new gadget calculations get added. I’d personally rip all this crap out.

@AzGilrock I fully agree with you. The code is littered with all kinds of bells-n-whistles that are of questionable value and come at a cost of performance. I would agree with/recommend either: 1) Removing the indicator entirely. OR  2) Make the indicator useful and intuitive. I don't think we should keep something that is confusing for most users.

**Comment by derwin12** [comment]
I smell a preference setting coming!

**Comment by AzGilrock** [comment]
When I see code changes I try to think about how often is this code going to run and will it slow down the GUI.   I remember when developed a lot of the grid code I would test the responsiveness.   Then I would wonder if it’s worse on an average computer since I usually had a work horse.   I just hope people think about those things and test whether speed is affected.

**Comment by computergeek1507** [comment]
I think this kind of indicator is pointless as 90% of users will have no idea what it means

**Comment by robfallone** [comment]
@AzGilrock and @computergeek1507, I can't agree with you more. I think having xLights be more performant is more important than this indicator. However, I would say if we are going to have the indicator it should be intuative and useful. Therfore, I strongly recommend we do either of the following 2 options but leaving it the way it is kinda the worst of both (having un-intutive feature that hurts performance).

1. Remove the indicator entirely with https://github.com/smeighan/xLights/pull/4076
2. Improve the indicator with https://github.com/smeighan/xLights/pull/4071. BTW, this improvement also improves performance because we removed a painful O(n2) linear search (i.e. ModelInView https://github.com/smeighan/xLights/pull/4071/files#r1368967392) from the code.
3. [Not Recommended] Leave the code as is. Keeping the un-intuitive feature while also degrading performance.

**Comment by derwin12** [comment]
I think this has been beat to death :-)
